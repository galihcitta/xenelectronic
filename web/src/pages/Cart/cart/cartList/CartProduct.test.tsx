import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CartProductProps, _CartProduct } from './CartProduct.component';

describe('<CartProduct />', () => {
  const mockAddItem = jest.fn();
  const mockRemoveItem = jest.fn();
  const mockClearItemFromCart = jest.fn();

  const props: CartProductProps = {
    addItem: mockAddItem,
    removeItem: mockRemoveItem,
    clearItemFromCart: mockClearItemFromCart,
    cartItemData: {
      character: 'modern',
      collection: 'clocks',
      color: 'black',
      description: 'lorem ipsum',
      id: '1',
      images: ['aa', 'bb'],
      name: 'Clock',
      price: '200',
      type: 'clocks'
    }
  };

  const setup = (props: CartProductProps) => {
    const utils = render(<_CartProduct {...props} />);
    const addBtn = utils.getByTestId('add-btn');
    const removeBtn = utils.getByTestId('remove-btn');
    const clearBtn = utils.getByTestId('clear-btn');

    return { ...utils, addBtn, removeBtn, clearBtn };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls `addItem` action', () => {
    const { addBtn } = setup(props);
    fireEvent.click(addBtn);

    expect(mockAddItem).toHaveBeenCalledTimes(1);
  });

  it('calls `removeItem` action', () => {
    const { removeBtn } = setup(props);
    fireEvent.click(removeBtn);

    expect(mockRemoveItem).toHaveBeenCalledTimes(1);
  });

  it('calls `clearItemFromCart` action', () => {
    const { clearBtn } = setup(props);
    fireEvent.click(clearBtn);

    expect(mockClearItemFromCart).toHaveBeenCalledTimes(1);
  });
});
