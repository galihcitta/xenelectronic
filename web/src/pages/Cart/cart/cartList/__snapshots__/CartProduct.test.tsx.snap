// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartProduct /> renders correctly 1`] = `
<DocumentFragment>
  <li
    class="sc-bdVaJa UQcAb"
  >
    <figure
      class="sc-bwzfXH djsJkE"
    >
      <img
        alt="Clock"
        class="sc-htpNat ElcCo"
        src="aa"
      />
    </figure>
    <div
      class="sc-bxivhb ellZpO"
    >
      <p
        class="sc-ifAKCX iTFAKT"
      >
        <strong>
          Clock
        </strong>
      </p>
      <p
        class="sc-ifAKCX iTFAKT"
      >
        <strong>
          $ 200
        </strong>
      </p>
      <p
        class="sc-ifAKCX iTFAKT"
      >
        Color: black
      </p>
      <div
        class="sc-EHOje iMsWhQ"
      >
        <button
          class="sc-bZQynM hdzTmk"
          data-testid="remove-btn"
        >
          -
        </button>
        <button
          class="sc-bZQynM hdzTmk"
          data-testid="add-btn"
        >
          +
        </button>
      </div>
    </div>
    <button
      class="sc-gzVnrw dLYbAw"
      data-testid="clear-btn"
    >
      x
    </button>
  </li>
</DocumentFragment>
`;
